Detections/OktaSSO/PasswordSpray.yaml

```diff
@@ -19,17 +19,17 @@ query: |
 
   let timeframe = ago(1h);
   let FailureThreshold = 15;
-  let LockOutEvents = Okta_CL
+  let FailedEvents = Okta_CL
   | where TimeGenerated > timeframe
-  | where eventType_s =~ "user.account.lock"
-  | summarize count() by client_ipAddress_s, bin(TimeGenerated, 5m)
-  | where count_ > FailureThreshold
-  | project TimeGenerated;
+  | where eventType_s =~ "user.session.start"and outcome_reason_s in ("VERIFICATION_ERROR","INVALID_CREDENTIALS")
+  | summarizedcount(actor_alternateId_s) by client_ipAddress_s, bin(TimeGenerated, 5m)
+  | where dcount_actor_alternateId_s > FailureThreshold
+  | project client_ipAddress_s, TimeGenerated;
   Okta_CL
   | where TimeGenerated > timeframe
-  | where eventType_s =~ "user.account.lock"
-  | extend TimeGenerated = bin(TimeGenerated, 5m)
-  | join kind=inner LockOutEvents () on TimeGenerated
-  | summarize User = make_set(actor_alternateId_s) by ClientIP = client_ipAddress_s, City = client_geographicalContext_city_s, Country = client_geographicalContext_country_s, bin(TimeGenerated, 5m)
-  | sort by TimeGenerated desc
-  | extend timestamp = TimeGenerated, IPCustomEntity = ClientIP
+  | where eventType_s =~ "user.session.start"and outcome_reason_s in ("VERIFICATION_ERROR","INVALID_CREDENTIALS")
+  | summarize Users = make_set(actor_alternateId_s) by client_ipAddress_s, City = client_geographicalContext_city_s, Country = client_geographicalContext_country_s, bin(TimeGenerated, 5m)
+  | joinkind=inner FailedEvents () on client_ipAddress_s, TimeGenerated
```

Comment:
Space needed between join and kind. 
Might also be a good idea to put FailedEvents  in the bracket i.e. like below
| join kind=inner (FailedEvents)  on client_ipAddress_s, TimeGenerated

Reply:
revisions submitted